x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix.tsx

```diff
@@ -0,0 +1,313 @@
+      }
+    }
+
+    const options = { didCancel: false };
+    fetchSplom(options);
+    return () => {
+      options.didCancel = true;
+    };
+  }, [fetchSize, fields, index, randomizeQuery, resultsField]);
+
+  const htmlId = useMemo(() => htmlIdGenerator()(), []);
+
+  useEffect(() => {
+    if (splom === undefined) {
+      return;
+    }
+
+    const { items, columns: rawColumns } = splom;
+
+    // Vega doesn't support dashes in attribute names
+    const columns = rawColumns.map((column) => `${column.split('-').join('_')}`);
+
+    const values =
+      resultsField !== undefined
+        ? items
+        : items.map((d) => {
+            d[`${resultsField}.${OUTLIER_SCORE_FIELD}`] = 0;
+            return d;
+          });
+
+    const vegaSpec = getScatterplotMatrixVegaLiteSpec(
+      values,
+      columns,
+      resultsField,
+      color,
+      legendType,
+      dynamicSize
+    );
+
+    const vgSpec = compile(vegaSpec).spec;
+
+    const view = new View(parse(vgSpec))
+      .logLevel(Warn)
+      .renderer('canvas')
+      .tooltip(new Handler().call)
+      .initialize(`#${htmlId}`);
+
+    view.runAsync(); // evaluate and render the view
+  }, [resultsField, splom, color, legendType, dynamicSize]);
+
+  return (
+    <>
+      {splom === undefined && (
+        <EuiText textAlign="center">
+          <EuiSpacer size="l" />
+          <EuiLoadingSpinner size="l" />
+          <EuiSpacer size="l" />
+        </EuiText>
+      )}
+      {splom !== undefined && (
+        <>
+          <EuiFlexGroup>
+            <EuiFlexItem>
+              <EuiFormRow
+                label={i18n.translate('xpack.ml.splom.fieldSelectionLabel', {
+                  defaultMessage: 'Fields',
+                })}
+                display="rowCompressed"
+                fullWidth
+              >
+                <EuiComboBox
+                  compressed
+                  fullWidth
+                  placeholder={i18n.translate('xpack.ml.splom.fieldSelectionPlaceholder', {
+                    defaultMessage: 'Select fields',
+                  })}
+                  options={fieldOptions}
+                  selectedOptions={fields.map((d) => ({
+                    label: d,
+                  }))}
+                  onChange={fieldsOnChange}
+                  isClearable={true}
+                  data-test-subj="mlScatterplotMatrixFieldsComboBox"
+                />
+              </EuiFormRow>
+            </EuiFlexItem>
+            <EuiFlexItem style={{ width: '200px' }} grow={false}>
+              <EuiFormRow
+                label={i18n.translate('xpack.ml.splom.SampleSizeLabel', {
+                  defaultMessage: 'Sample size',
+                })}
+                display="rowCompressed"
+                fullWidth
+              >
+                <EuiSelect
+                  compressed
+                  options={sampleSizeOptions}
+                  value={fetchSize}
+                  onChange={(e) => fetchSizeOnChange(e)}
```

Comment:
```suggestion
                  onChange={fetchSizeOnChange}
```

Reply:
Changed in c1591bf. (Note I copied this from EUI's examples :)).